### PR TITLE
pkg_resources: Clarify some methods return `bytes`, not `str`

### PR DESCRIPTION
**Pull request by Avasam**
## Summary of changes

Clarify some `pkg_resources` methods return `bytes`, not `str`. Also return an empty `bytes` in `EmptyProvider._get`

Works towards #2345 and merging `pkg_resources` stubs from typeshed

### Pull Request Checklist
- [ ] Changes have tests (`EmptyProvider._get` doesn't have tests. type-checking tests are coming in an upcoming PR)
- [x] News fragment added in [`newsfragments/`].
  _(See [documentation][PR docs] for details)_

[`newsfragments/`]: https://github.com/pypa/setuptools/tree/master/newsfragments
[PR docs]:
https://setuptools.pypa.io/en/latest/development/developer-guide.html#making-a-pull-request

### CODE REVIEW

File: newsfragments/4243.bugfix.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Clarify some `pkg_resources` methods return `bytes`, not `str`. Also return an empty `bytes` in `EmptyProvider._get` -- by :user:`Avasam`
```
Comment (Avasam) [inline newsfragments/4243.bugfix.rst:1]:
```suggestion
Clarify some `pkg_resources` methods return `bytes`, not `str`. Also return an empty `bytes` in ``EmptyProvider._get`` -- by :user:`Avasam`
```

File: pkg_resources/__init__.py [hunk 2]
```diff
@@ -566,8 +566,8 @@ def get_resource_stream(self, manager, resource_name):
 
         `manager` must be an ``IResourceManager``"""
 
-    def get_resource_string(self, manager, resource_name):
-        """Return a string containing the contents of `resource_name`
+    def get_resource_string(self, manager, resource_name) -> bytes:
+        """Return a bytes string containing the contents of `resource_name`
```
Comment (abravalheri) [inline pkg_resources/__init__.py:570]:
```suggestion
        """Return the contents of `resource_name` as :obj:`bytes`
```

File: pkg_resources/__init__.py [hunk 3]
```diff
@@ -1203,8 +1203,8 @@ def resource_stream(self, package_or_requirement, resource_name):
             self, resource_name
         )
 
-    def resource_string(self, package_or_requirement, resource_name):
-        """Return specified resource as a string"""
+    def resource_string(self, package_or_requirement, resource_name) -> bytes:
+        """Return specified resource as a bytes string"""
```
Comment (abravalheri) [inline pkg_resources/__init__.py:1207]:
```suggestion
        """Return specified resource as :obj:`bytes`"""
```

**Comment by abravalheri** [comment]
Thank you very much!

**Pull request merged**